### ISSUE TITLE
1.16.5 crash after reload

### ISSUE THREAD

**Issue by rlnt** [TinyModularThings/Scavenge#11]
Hey there,
I am getting a crash after reloading my modpack. Only reloading, joining the world works normally.
This only happens with Scavenge being installed. Uninstalling it again fixes it. Didn't change any configs yet.

Here is the crash log: https://pastebin.com/HuEK1sfZ

Versions
Minecraft: 1.16.5
Forge: 36.1.32
Scavenge: 2.0.2

**Comment by Speiger** [comment]
My JEI Hack is not compatible with Cooking for block heads. The issue is if you want JEI compat you kinda need to reload Scavenge twice. Otherwise it will not be loading any JEI data for scavenge.

To explain what my JEI hack does so the mods that crash with it can not say "Its scavenges bug he has to fix it" because it is also a problem with these two mods that this happens in the first place.

What i do is when i receive scavenge data on the client is to basically send a event out that tells that the Tags were updated. 
This just triggers that anything that uses tags to be reloaded on the client side with data that was already present.
But only on mods. Which should be technically fine since the data hasn't changed it just forces everything to reload.

The crash with Cooking for Blockheads & Mantle (Tinkers base mod) has is basically they can't deal with that tags got reloaded twice even so the data never changed.

The fix on my side is hack into JEI specifically and hope he never changes how JEI reloads.


TL:DR Scavenge throws a reload event a second time with exactly the same data that is present already to force jei to reload again and Mantle & Cooking for BlockHeads can not deal with that which they should, because otherwise they will have a lot of issues because JEI has no way to sync late data into JEI other then this.

**Comment by rlnt** [comment]
Well, that's unfortunate. This makes it useless for my purpose. Thanks for the detailed explanation though.

**Comment by Speiger** [comment]
I will try to fix the issue but this will take a update or two to be done. Sry

**Comment by Speiger** [comment]
Patch comes out this weekend and will fix the JEI stuff too
